app/src/test/java/com/group80/uoftinder/GenerateCompatibilityListUnitTest.java

```diff
@@ -0,0 +1,135 @@
+package com.group80.uoftinder;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.group80.uoftinder.create_account_use_case.CreateAccountInteractor;
+import com.group80.uoftinder.entities.User;
+import com.group80.uoftinder.feed.GenerateCompatibilityList;
+import com.group80.uoftinder.feed.RecommendationPresenter;
+import com.group80.uoftinder.feed.RecommendationView;
+import com.group80.uoftinder.feed.UserScoreFacade;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A GenerateCompatibilityListUnitTest class that tests the functionality of the
+ * GenerateCompatibilityList class
+ */
+public class GenerateCompatibilityListUnitTest {
+
+    /**
+     * Test that GenerateCompatibilityList.orderCompatibilityList reorders the
+     * compatibility list from most compatible user to least compatible user when there are
+     * 3 total users
+     */
+    @Test
+    public void orderCompatibilityListTest1() {
+        User curUser = new User("curUser");
+        int curUserScore = 11682;
+        curUser.setScore(curUserScore);
+//        genCompatibilityList.setCurUser(curUser);
+
+        CreateAccountInteractor.setAnswerSchema(new int[] {3, 4, 4, 5, 4});
+        CreateAccountInteractor.setIsMultiSelect(new boolean[] {false, true, true, false, false});
+
+        List<List<Integer>> userAnswers = new ArrayList<>();
+        userAnswers.add(Collections.singletonList(1));
+        userAnswers.add(Arrays.asList(1, 2));
+        userAnswers.add(Arrays.asList(0, 1, 3));
+        userAnswers.add(Collections.singletonList(0));
+        userAnswers.add(Collections.singletonList(2));
+        curUser.setAnswers(userAnswers);
+
+        GenerateCompatibilityList genCompatibilityList = new GenerateCompatibilityList(curUser);
```

Comment:
Is initializing ```GenerateCompatibilityList``` with a ```User``` object a permanent implementation? In that case, when a ```GenerateCompatibilityList``` object is initialized in the project, we need to pass current user in instead of getting it from the database.